exonum/src/runtime/mod.rs

```diff
@@ -336,6 +371,19 @@ pub trait Runtime: Send + fmt::Debug + 'static {
     /// The default implementation does nothing.
     fn initialize(&mut self, blockchain: &Blockchain) {}
 
+    /// Checks if the runtime supports an optional feature.
+    ///
+    /// This method can be called by the core before performing operations that might not
+    /// be implemented in a runtime, or by the supervisor service in order to check that a potential
+    /// service / artifact state transition can be handled by the runtime.
+    ///
+    /// An implementation should return `false` for all features the runtime does not recognize.
+    /// The default implementation always returns `false`, i.e., signals that the runtime supports
+    /// no optional features.
+    fn is_supported(&self, feature: &RuntimeFeature) -> bool {
```

Comment:
@dmitry-timofeev Here's the new method in the `Runtime` interface, through which the runtime can signal that it supports service freezing (in the future, this could be extended to other features, such as artifact unloading).

Reply:
I see, thanks! (cc @vitvakatu the things we don't support yet)